libsweep/config.mk

```diff
@@ -1,15 +1,81 @@
 VERSION_MAJOR = 0
 VERSION_MINOR = 1
 
-UNAME := $(shell uname)
+# project directory containing src files
+SRC_DIR := src
+# project directory containing include header files
+INC_DIR := inc
+# project directory containing sweep include header files
+SWEEP_INC_DIR := inc/sweep
 
+# created directory where object files will be placed
+OBJ_DIR := obj
+# created directory where outputs will be placed
+BIN_DIR := bin
+
+# Define some commands (these might need to change depending on platform)
+MKDIR_P := mkdir -p
+
+# read the operating system
+UNAME := $(shell uname -s)
+
+# Note if the uname indicates any form of MINGW32 on windows
+ifeq ($(findstring MINGW,$(UNAME)),MINGW)
+  #if UNAME contains MINGW32
+  UNAME = MINGW
+else ifeq ($(findstring MSYS,$(UNAME)),MSYS)
+  #if UNAME contains MSYS
+  UNAME = MSYS
+endif
+
+# Set Options & Flags by Operating System
 ifeq ($(UNAME), Linux)
-  PREFIX ?= /usr
+  # For linux platforms
+  target = libsweep.so
+  dummy_target = dummy_linux
+  SRC_ARCH_DIR := src/arch/unix
+  INSTALL_DIR_LIB ?= /usr/lib
+  INSTALL_DIR_INCLUDES ?= /usr/include/sweep
+  LINKER = g++
```

Comment:
I was passing CXXFLAGS during the linking process as well. As i understood it, when specifying the `-shared` linker option it was safer to include the compilation options as well. I tried messing around a bit with getting ld to work, but my experimentation was introducing more complexity to the config and makefile. If you have a good fix for this on linux within the new config/makefile context I'm happy to test and adopt it for the windows options as well.